plugins/modules/data_context.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerMutableModule,
+)
+from cm_client import DataContextsResourceApi, ApiDataContextList
+
+from cm_client import (
+    ClustersResourceApi,
+    ApiDataContext,
+)
+from cm_client.rest import ApiException
+from ansible_collections.cloudera.cluster.plugins.module_utils.data_context_utils import (
+    parse_data_context_result,
+)
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: data_context
+short_description: Create, update, or delete a data context
+description:
+  - Configure details of a specific data context.
+  - Create a new data context.
+  - Update an existing data context.
+  - Delete a data context.
+  - The module supports C(check_mode).
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm-client >= 54
+options:
+  name:
+    description:
+      - The name of the data context.
+    type: str
+    required: yes
```

Comment:
For consistency with the other modules in the collection, `name` and `cluster` should have `aliases`, i.e. `context_name`, `data_context_name`, and `cluster_name`.